### ISSUE TITLE
Unable to use the latest riverpod in flutter version lower than 3.16

### ISSUE THREAD

**Issue by kefasjw** [rrousselGit/riverpod#3574]
**Describe the bug**

Riverpod uses the collection package version ^1.18.0, but flutter pins the collection package version to 1.17.2. Therefore, the latest stable riverpod (2.5.1) can't be used in flutter version lower than 3.16

**To Reproduce**

- Install flutter version 3.13 or lower
- Add `flutter_riverpod: ^2.5.1` dependency to a `pubspec.yaml`
- Run `flutter pub get`

And this error shows up:
```
Note: collection is pinned to version 1.17.2 by flutter from the flutter SDK.
See https://dart.dev/go/sdk-version-pinning for details.


Because every version of flutter from sdk depends on collection 1.17.2 and riverpod >=2.5.1 <3.0.0-dev.0 depends on collection ^1.18.0, flutter from sdk is
  incompatible with riverpod >=2.5.1 <3.0.0-dev.0.
And because flutter_riverpod >=2.5.1 <3.0.0-dev.0 depends on riverpod 2.5.1, flutter from sdk is incompatible with flutter_riverpod >=2.5.1 <3.0.0-dev.0.
So, because flutter_playground depends on both flutter from sdk and flutter_riverpod 2.5.1, version solving failed.
```

**Expected behavior**
flutter riverpod 2.5.1 should be able to be used in older flutter version lower than 3.16. Lower version of `collection` package should be used instead of 1.18.0.

**Comment by rrousselGit** [comment]
That's low priority for me. I'd rather have up-to-date dependencies than dealing with broad version ranges.

You can use `dependency_overrides` to bypass this.

**Comment by rrousselGit** [comment]
Also Riverpod is soon going to require Dart 3 anyway, which requires collection 1.17

So I don't think that's realistic

**Comment by sharpsan** [comment]
> That's low priority for me. I'd rather have up-to-date dependencies than dealing with broad version ranges.
> 
> You can use `dependency_overrides` to bypass this.

You should consider the fact that this creates a conflict between your package and the latest stable version of Flutter.

**Comment by rrousselGit** [comment]
The cost of trying to always support stable is: I have to maintain multiple versions of Riverpod in parallel from months.   
I get that this is painful, but the issue fixes itself with a bit of time. Or you can switch to the dev channel.

I'd rather spend time working on the package in a meaningful way than to spend it dealing with rebases & stuff just to maintain 2 concurrent versions.
That's also why the 3.0 dev release stopped getting updates. It was too tedious.

**Comment by sharpsan** [comment]
> The cost of trying to always support stable is: I have to maintain multiple versions of Riverpod in parallel from months. I get that this is painful, but the issue fixes itself with a bit of time. Or you can switch to the dev channel.
> 
> I'd rather spend time working on the package in a meaningful way than to spend it dealing with rebases & stuff just to maintain 2 concurrent versions. That's also why the 3.0 dev release stopped getting updates. It was too tedious.

That's valid.  I would say it's more confusing than it is painful, because while it makes sense to stay on an older version until other things catch up, this isn't intuitive.  Even a note somewhere in the readme or changelog would help, specifying expected conflicts between riverpod and Flutter stable.

**Comment by AhmedLSayed9** [comment]
Which riverpod version conflicts with the latest stable version of Flutter? 👀 
I'm using latest stable version from both.

**Comment by sharpsan** [comment]
> Which riverpod version conflicts with the latest stable version of Flutter? 👀 I'm using latest stable version from both.

Do you have riverpod_lint as a dep?

**Comment by AhmedLSayed9** [comment]
> > Which riverpod version conflicts with the latest stable version of Flutter? 👀 I'm using latest stable version from both.
> 
> Do you have riverpod_lint as a dep?

yeah true, I'm using riverpod_lint `v2.3.10`. I guess `dependency_overrides` is the way to go If you need to use earlier versions from riverpod_lint.
